File: segundos 4 lenguajes/Java/JugadorAtaque.java
package Java;
abstract class JugadorAtaque {
    String nombre;
    int nivel;

    public JugadorAtaque() {}

    public JugadorAtaque(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorAtaque(int nivel) {
        this.nombre = "Anon";
        this.nivel = nivel;
    }

    public JugadorAtaque(JugadorAtaque j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }

    abstract void golpear();
}

File: segundos 4 lenguajes/js/JugadorAtaque.js
export class JugadorAtaque {
    constructor(nombre = "Anon", nivel = 1, copia = null) {
        if (copia) {
            this.nombre = copia.nombre;
            this.nivel = copia.nivel;
        } else {
            this.nombre = nombre;
            this.nivel = nivel;
        }
    }

    golpear() {
        throw new Error("Método golpear() debe implementarse.");
    }
}
